wiredtiger/mongo-tests
# WT-11603 tests

Tests for WT-11603 https://github.com/wiredtiger/wiredtiger/pull/10031.

* All tests live in the `WT-11603` subdirectory.
* Run tests by running `run.sh` or `run-mirror.sh`.
* It will download the required sources, patch them and run tests.
* Tests include:
  * Simple test that it works by calling a packing and unpacking function.
  * Running an example with the new code.
  * Running every packing and unpacking function with various inputs and checking the result.
  * Garbage test: no garbage can break the unpacker and it will return a correct error code.
  * Mirror/shadow test:
    * Every packing and unpacking function is replaced with one that would call both (old and new) functions and make sure their outputs are identical.
    * Run all wt tests.
    * Run test/format with oplog.
    * Run all evergreen variants/tests.
